# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/mathieu/ocaml/tcltk.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ru ocaml-3.06.old/otherlibs/labltk/support/cltkImg.c ocaml-3.06/otherlibs/labltk/support/cltkImg.c
--- ocaml-3.06.old/otherlibs/labltk/support/cltkImg.c	2002-07-23 16:11:59.000000000 +0200
+++ ocaml-3.06/otherlibs/labltk/support/cltkImg.c	2003-03-20 21:03:34.000000000 +0100
@@ -99,7 +99,11 @@
   pib.offset[0] = 0;
   pib.offset[1] = 1;
   pib.offset[2] = 2;
-  Tk_PhotoPutBlock(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)); 
+  Tk_PhotoPutBlock(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)
+#if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
+                   , TK_PHOTO_COMPOSITE_SET
+#endif
+    ); 
 }
 
 CAMLprim void camltk_setimgdata_bytecode(argv,argn)
diff -ru ocaml-3.06.old/otherlibs/labltk/tkanim/tkAnimGIF.c ocaml-3.06/otherlibs/labltk/tkanim/tkAnimGIF.c
--- ocaml-3.06.old/otherlibs/labltk/tkanim/tkAnimGIF.c	2002-07-23 16:12:00.000000000 +0200
+++ ocaml-3.06/otherlibs/labltk/tkanim/tkAnimGIF.c	2003-03-20 21:04:15.000000000 +0100
@@ -335,8 +335,11 @@
                 goto error;
             }
         }
-        Tk_PhotoPutBlock(photoHandle, &block, 0, 0, 
-                         imageWidth, imageHeight);
+        Tk_PhotoPutBlock(photoHandle, &block, 0, 0, imageWidth, imageHeight
+#if (TK_MAJOR_VERSION == 8 && TK_MINOR_VERSION >= 4 || TK_MAJOR_VERSION > 8)
+                   , TK_PHOTO_COMPOSITE_SET
+#endif
+          );
 #ifdef TKANIM_DEBUG
     fprintf(stderr, " Retrieving result\n");
 #endif
